### PR TITLE
Remove ClusterCSIDriver.Spec.DriverConfig

### PR DESCRIPTION
**Pull request by jsafrane**
We haven't found any use of the structure. It's better remove it now, as we can add it in future, but we can't ever remove it.

### CODE REVIEW

**Comment by deads2k** [comment]
/lgtm
/hold

holding in case @jsafrane has second thoughts.  If added in the future, consumers need to understand what to do when it is missing and handle upgrade/downgrade data loss issues.

@jsafrane feel free to remove the hold at your discretion.

File: operator/v1/types_csi_cluster_driver.go [hunk 1]
```diff
@@ -35,40 +33,16 @@ type ClusterCSIDriver struct {
...
-type CSIDriverName string
-
-// If you are adding a new driver name here, ensure that kubebuilder:validation:Enum is updated above
-// and 0000_90_cluster_csi_driver_01_config.crd.yaml-merge-patch file is also updated with new driver name.
-const (
-	AWSEBSCSIDriver CSIDriverName = "ebs.csi.aws.com"
-	ManilaCSIDriver CSIDriverName = "manila.csi.openstack.org"
-	OvirtCSIDriver  CSIDriverName = "csi.ovirt.org"
-)
-
```
Comment (gnufied) [inline operator/v1/types_csi_cluster_driver.go:49]: Couple of things as discussed in standup. 

1. We need these constants elsewhere and deleting them here will cause that code to break. Do we want to keep these constants here, so as each operator does not need to define individually?

2. Also - a casual reader of code may not understand, how different types can be disambiguated now. The fact that, they are disambiguated based on name sits outside this code and is not obvious. We need to document this at very minimum.

Comment (gnufied) [reply operator/v1/types_csi_cluster_driver.go:49]: cc @deads2k ^

Comment (jsafrane) [reply operator/v1/types_csi_cluster_driver.go]: 1. I restored the constants and `CSIDriverName` type.
2. I added "Name of the object must be name of the CSI driver it operates. See CSIDriverName type for list of allowed values." to `ClusterCSIDriver` description

File: operator/v1/types_csi_cluster_driver.go [hunk 2]
```diff
@@ -12,15 +12,14 @@ import (
...
-//       driverName: "ebs.csi.aws.com"
 
 // +genclient
 // +genclient:nonNamespaced
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
 // ClusterCSIDriver object allows management and configuration of a CSI driver operator
-// installed by default in OpenShift.
+// installed by default in OpenShift. Name of the object must be name of the CSI driver
+// it operates. See CSIDriverName type for list of allowed values.
```
Comment (bertinatto) [inline operator/v1/types_csi_cluster_driver.go:22]: > See CSIDriverName type for list of allowed values.
Down below we remove the list of allowed values.

Comment (jsafrane) [reply operator/v1/types_csi_cluster_driver.go:22]: if you mean `kubebuilder` comment, then that one would not be used anywhere, as there is no field with CSIDriverName type.

Comment (bertinatto) [reply operator/v1/types_csi_cluster_driver.go:22]: Meant to say that I didn't find any "list of allowed values" cited in the comment.

Comment (bertinatto) [reply operator/v1/types_csi_cluster_driver.go:22]: oh, nevermind; I found it

**Pull request merged**